src/components/index.cjsx

```diff
@@ -20,15 +20,26 @@ Dashboard = React.createClass
   displayName: 'Dashboard'
   componentWillMount: -> CurrentUserStore.addChangeListener(@update)
   componentWillUnmount: -> CurrentUserStore.removeChangeListener(@update)
+  
+  contextTypes:
+    router: React.PropTypes.func
 
   update: -> @setState({})
+
   render: ->
     if CurrentUserStore.isCoursesLoaded()
       courses = CurrentUserStore.getCourses()
+      if courses.length is 1
+        roles = courses[0].roles
+        if roles.length is 1
+          singleCourseId = courses[0].id
+          if roles[0].type is 'student'
+            @context.router.transitionTo('viewStudentDashboard', {courseId: singleCourseId})
```

Comment:
How important is it that the url always include "/dashboard" when what is displayed is the Dashboard?  

I'm thinking it might make sense to replace the task list with a shell which employ the above logic to decide whether to render either the task listing or the student dashboard.

That would remove the need for a redirect and the url would remain the same.  Only what is displayed would change depending on the number of courses/roles.